test/scripts/common.inc

```diff
@@ -0,0 +1,63 @@
+#
+# Include file with common parts shared for local and ephemeral
+#
+
+# Troubleshooting:
+# - Run with DEBUG=1 to see some traces from curl.sh wrapper
+#   $ DEBUG=1 ./test/scripts/local-domain-token.sh
+# - Run with more verbose by:
+#   $ DEBUG=1 bash -xv ./test/scripts/local-domain-token.sh
+
+set -eo pipefail
+
+function error {
+    local err=$?
+    printf "ERROR: %s\n" "$*" >&2
+    exit $err
+}
+
+ORG_ID="${ORG_ID:-12345}"
+
+export IDENTITY_USER=""   # Use $(identity_user)
+export IDENTITY_SYSTEM="" # Use $(identity_system)
+export IDM_VERSION=""     # Use $(idm_version)
+
+SRCDIR="$(dirname "${BASH_SOURCE[0]}")"
+# shellcheck disable=SC2034
+BASEDIR="$(dirname "$(dirname "${SRCDIR}")")"
+REPOBASEDIR="$(git rev-parse --show-toplevel)"
+export REPOBASEDIR
+export XRHIDGEN="${REPOBASEDIR}/tools/bin/xrhidgen"
+
+if [[ ! -x "${XRHIDGEN}" ]]; then
+    error "${XRHIDGEN} is missing, run 'make install-tools'"
+    exit 2
+fi
+
+function identity_user() {
+    if [ "${IDENTITY_USER}" == "" ]; then
+        IDENTITY_USER="$( "${XRHIDGEN}" -org-id "${ORG_ID}" user -is-active=true -is-org-admin=true -user-id test -username test | base64 -w0 )"
+        [ "${IDENTITY_USER}" != "" ] || error "IDENTITY_USER is empty"
+    fi
+    printf "%s" "${IDENTITY_USER}"
+}
```

Comment:
The function looks unnecessarily complicated. Why do you cache the value? What is wrong with:

```suggestion
function identity_user() {
    "${XRHIDGEN}" -org-id "${ORG_ID}" user -is-active=true -is-org-admin=true -user-id test -username test | base64 -w0}
```

Reply:
nothing is wrong, I thought it could be better! but rethinking, if we are calling at some point with a different place with other ORG_ID, the returned output would be wrong; so yes, better do not cache and reduce complexity.